src/main.cpp

```diff
@@ -2812,15 +2812,17 @@ bool static DisconnectTip(CValidationState &state, bool fBare = false) {
     // Update chainActive and related variables.
     UpdateTip(pindexDelete->pprev);
     // Get the current commitment tree
-    ZCIncrementalMerkleTree newTree;
-    assert(pcoinsTip->GetSproutAnchorAt(pcoinsTip->GetBestAnchor(SPROUT), newTree));
+    ZCIncrementalMerkleTree newSproutTree;
+    ZCSaplingIncrementalMerkleTree newSaplingTree;
+    assert(pcoinsTip->GetSproutAnchorAt(pcoinsTip->GetBestAnchor(SPROUT), newSproutTree));
+    assert(pcoinsTip->GetSaplingAnchorAt(pcoinsTip->GetBestAnchor(SAPLING), newSaplingTree));
```

Comment:
I like that idea. It would be nice to do some refactoring around some of the sprout v sapling logic and I think this would be a step in the right direction.

Reply:
@Eirik0 Can you file a Github issue about this refactoring?  It can be done later.

Reply:
https://github.com/zcash/zcash/issues/3403